sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -572,8 +573,13 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
 
   private def assertNotBucketed(operation: String): Unit = {
     if (numBuckets.isDefined || sortColumnNames.isDefined) {
-      throw new IllegalArgumentException(
```

Comment:
@cloud-fan, would you mind if this patch changes `IllegalArgumentException` here to `AnalysisException`? Since in most places in this class we throw `AnalysisException`, :-)

This change should be OK as this has not been actually in any non-preview releases.